src/core/main.c

```diff
@@ -3096,6 +3096,12 @@ int main(int argc, char *argv[]) {
         __lsan_do_leak_check();
 #endif
 
+        /* If we're running in a container, we can communicate the exit status normally. When we're running
+         * in a VM, that's not possible so we propagate the exit status via the notify socket instead if one
+         * is configured. */
+        if (detect_container() == 0)
```

Comment:
This will send the notify also on baremetal, right? Is that something we want to avoid or will it be harmless?

Reply:
I figured it'd be harmless? You could set vmm.notify_socket and systemd would send something to it on bare metal, but it wouldn't hurt either, and we'd already be sending READY=1 to that as wel